tests/unit/moduleapi/list.tcl

```diff
@@ -64,6 +64,15 @@ start_server {tags {"modules"}} {
         r list.getall k
     } {bar y foo}
 
+    test {Module list - Crash due to reuse iterator entry after deletion} {
```

Comment:
I think we are testing that it *doesn't crash* (the solution), not that it crashes (the problem).